File: 18학번_김남희/week02/BJ02740_MatrixMult.py
#!/usr/bin/env python
# coding: utf-8

# In[ ]:


"""
201824421 Kim Namhee
Baekjoon 02740 : Matrix Multiplication - Get C = A * B
Language : Python 3
Code Length : 438 B, Time : 248 ms, Mem : 29,380 KB
"""


# In[ ]:


# Set array A
n, m = map(int, input().split())
a = []
for _ in range(n):
    a.append(list(map(int, input().split())))


# In[3]:


# Set array B
m, k = map(int, input().split())
b = []
for _ in range(m):
    b.append(list(map(int, input().split())))


# In[4]:


# Initail array C
c = [[0 for _ in range(k)] for _ in range (n)]


# In[5]:


# Calculate A*B = C
for i in range(n):
    for j in range(k):
        for l in range(m):
            c[i][j] += a[i][l] * b[l][j]


# In[6]:


# Print array C
for i in c:
    for j in i:
        print(j, end = ' ')
    print()

File: 18학번_김남희/week02/BJ02740_MatrixMult.cpp
/* 201824421 Kim Namhee
 *	Baekjoon 02740 : Matrix Multiplication - Get C = A * B
 *	Language : Python 3
 *	Code Length : 578 B, Time : 8 ms, Mem : 1,984 KB
 */

#include <iostream>

using namespace std;

int main() {
	int n, m, k;
	cin >> n >> m;
	
	// Set array A
	int A[n][m];
	
	for (int i = 0; i < n; i++) 
		for (int j = 0; j < m; j++)
			cin >> A[i][j];
	
	cin >> m >> k;
	
	// Set array B
	int B[m][k];
	
	for (int i = 0; i < m; i++) 
		for (int j = 0; j < k; j++)
			cin >> B[i][j];

	// Set array C
	int C[n][k];

	for (int i = 0; i < n; i++) 
		for (int j = 0; j < k; j++)
			C[i][j] = 0;
	
	for (int i = 0; i < n; i++) {
		for (int j = 0; j < k; j++) {
			for (int h = 0; h < m; h++) {
				C[i][j] += A[i][h] * B[h][j];
			}
			// print array C
			cout << C[i][j] << " ";
		}
		cout << endl;
	}
}
